public/app/features/scenes/variables/sceneTemplateInterpolator.ts

```diff
@@ -7,6 +7,11 @@ import { SceneObject } from '../core/types';
 import { SceneVariable } from './types';
 
 export function sceneTemplateInterpolator(target: string, sceneObject: SceneObject) {
+  // Skip any interpolation if there are no variables in the scene object graph
+  if (!sceneObject.getVariables()) {
+    return target;
+  }
```

Comment:
As a matter of consistency I think `getVariables` should throw similarly to how `getData` and `getTimeRange` do. Then we could just do :
```ts
try {
  const vars = sceneObject.getVariables();
  ...
} catch (e) {
  console.error(e);
  return target;
}

Reply:
I am torn, when add support for non data viz panels (like text / news) we have to remove that throw from getData.  I ran into this problem in https://github.com/grafana/grafana/pull/58591/

think we should change getData to something like

```ts
  getData(): SceneObject<SceneDataState> {
    const { $data } = this.state;
    if ($data) {
      return $data;
    }

    if (this.parent) {
      return this.parent.getData();
    }

    return EmptyDataNode;
  }
```
 
Where EmptyDataNode is a const global

Reply:
Or we remove getData, getTimeRange and getVariables by external graphWalker util functions?

Reply:
Personally, I really like the getData/TimeRange/Variables interface directly on the SceneObject interface. `graphWalker util functions` feels funky to me, yet another player in the game.

So, with the EmptyDataNode it seems good, but what would that be for the time range? A default time range? Meaning that every scene comes with a time range that one can override on the root level? Maybe this is a good idea.